### PR TITLE
statistics: add docs for version 2

### PR DESCRIPTION
**Pull request by winoros**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Add descption about the analyze's changes based on the variable `tidb_analyze_version`

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: 
- Other reference link(s):https://github.com/pingcap/tidb/pull/20580 https://github.com/pingcap/tidb/pull/21817

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: statistics.md [hunk 1]
```diff
@@ -290,6 +313,31 @@ SHOW STATS_BUCKETS [ShowLikeOrWhere];
 | repeats | 最大值出现的次数 |
 | lower_bound | 最小值 |
 | upper_bound | 最大值 |
+| ndv | 不同值的个数，当 analyze version 为 1 时，该值没有实际意义，因此会有 count 不为 0，但 ndv 为 0 的情况 |
```
Comment (time-and-fate) [inline statistics.md:316]: 直接说 version 为 1 时恒为 0 没有意义是不是会清晰一些？

File: statistics.md [hunk 2]
```diff
@@ -5,7 +5,25 @@ aliases: ['/docs-cn/dev/statistics/','/docs-cn/dev/reference/performance/statist
...
 # 统计信息简介
 
-在[索引的选择](/choose-index.md)章节中，提到了 TiDB 会使用统计信息来决定选择哪个索引。在 TiDB 中，我们维护的统计信息包括表的总行数，列的等深直方图，Count-Min Sketch，Null 值的个数，平均长度，不同值的数目等等。本文将简单介绍直方图和 Count-Min Sketch，以及详细介绍统计信息的收集和维护。
+在[索引的选择](/choose-index.md)章节中，提到了 TiDB 会使用统计信息来决定选择哪个索引。在 TiDB 中，我们会使用变量 `tidb_analyze_version` 来控制 TiDB 所收集到的的统计信息。目前 TiDB 中支持 `tidb_analyze_version = 1` 以及 `tidb_analyze_version = 2` 两种统计信息，其中值为 1 是当前的默认值。目前两种版本中，TiDB 维护的统计信息可以见如下的表格：
+
+| 信息 | Version 1 | Version 2|
+| --- | --- | ---|
+| 表的总行数 | √ | √ |
+| 列的Count-Min Sketch | √ | × |
+| 索引的Count-Min Sketch | √ | × |
```
Comment (time-and-fate) [inline statistics.md:14]: 这里似乎中英文之间要空格

File: statistics.md (hunk 1 above)
Comment (qw4990) [reply statistics.md:316]: +1

**Review by qw4990 — COMMENTED** [review]
LGTM, please address existed comments @winoros

File: statistics.md [hunk 3]
```diff
@@ -5,7 +5,25 @@ aliases: ['/docs-cn/dev/statistics/','/docs-cn/dev/reference/performance/statist
...
+| 列的 Top-N | √ | √（改善了维护方式和精度） |
+| 索引的 Top-N | √(维护精度不足，不能用于估计) | √ |
```
Comment (time-and-fate) [inline statistics.md:16]: 这个表述是不是有点奇怪，毕竟 version 1 也是有在使用 topn 做估算的吧？
要不要和列 topn 换成一样的表述，直接说改善了维护方式和精度？

**Review by time-and-fate — COMMENTED** [review]
LGTM

File: statistics.md [hunk 4]
```diff
@@ -5,7 +5,25 @@ aliases: ['/docs-cn/dev/statistics/','/docs-cn/dev/reference/performance/statist
...
+| 列的 Top-N | √ | √（改善了维护方式和精度） |
+| 索引的 Top-N | √(维护精度不足，会产生较大误差) | √（改善了维护方式和精度） |
+| 列的直方图 | √ | √（直方图中不包含 Top-N 中出现的值） |
+| 索引的直方图 | √ | √（直方图的桶中记录了各自的不同值的个数，且直方图不包含 Top-N 中出现的值） |
+| 列的 NULL 值个数 | √ | √ |
+| 索引的 NULL 值个数 | √ | √ |
+| 列的平均长度 | √ | √ |
+| 索引的平均长度 | √ | √ |
+
+Version 2 的统计信息避免了 Version 1 中因为哈希冲突导致的在较大的数据量中可能产生的较大误差，提高了大多数场景中的估算精度。
```
Comment (ran-huang) [inline statistics.md:24]: These paragraphs should be put in **a separate section** called "统计信息分类" or "统计信息简介".

File: statistics.md [hunk 5]
```diff
@@ -290,6 +313,31 @@ SHOW STATS_BUCKETS [ShowLikeOrWhere];
 | repeats | 最大值出现的次数 |
 | lower_bound | 最小值 |
 | upper_bound | 最大值 |
+| ndv | 当前桶内不同值的个数。当 analyze version 为 1 时，该值恒为 0，没有实际意义。 |
```
[the remark this answers is not in the document]
Comment (winoros) [reply statistics.md:316]: Yes

**Review by ran-huang — APPROVED** [review]

**Pull request merged**